src/engine/effects/engineeffectchain.cpp

```diff
@@ -200,7 +197,7 @@ void EngineEffectChain::process(const ChannelHandle& handle,
     CSAMPLE* pIntermediateOutput = m_buffer1.data();
 
     for (EngineEffect* pEffect: m_effects) {
-        if (pEffect == NULL || !pEffect->enabled()) {
+        if (pEffect == nullptr || pEffect->disabled()) {
             continue;
         }
         pEffect->process(
```

Comment:
This call needs to be made for the intermediate disabling/enabling EffectProcessor states. Previously the optimization was returning from the function when the chain was fully dry before getting to this line.